casper-private-auction-core/src/auction.rs

```diff
@@ -96,23 +102,46 @@ impl Auction {
             match caller {
                 CallStackElement::StoredContract {
                     contract_package_hash,
-                    contract_hash: _,
-                } => Key::Hash(contract_package_hash.value()),
-                _ => runtime::revert(AuctionError::InvalidCaller),
+                    contract_hash,
+                } => (
+                    Some(Key::Hash(contract_hash.value())),
+                    Some(Key::Hash(contract_package_hash.value())),
+                ),
+                _ => (None, None),
             }
+        }) else {
+            runtime::revert(AuctionError::InvalidCaller);
         };
-        let mut token_ids = alloc::vec::Vec::new();
-        token_ids.push(AuctionData::get_token_id());
-        runtime::call_versioned_contract(
-            AuctionData::get_nft_hash(),
-            None,
-            "transfer",
-            runtime_args! {
-              "sender" => auction_key,
-              "recipient" => recipient,
-              "token_ids" => token_ids,
-            },
-        )
+
+        let has_enhanced_nft: bool = AuctionData::get_has_enhanced_nft();
+        let token_id: String = AuctionData::get_token_id();
+        if !has_enhanced_nft {
+            // CEP-47
+            let token_ids = vec![token_id];
+            runtime::call_versioned_contract::<()>(
+                AuctionData::get_nft_hash(),
+                None,
+                TRANSFER,
+                runtime_args! {
+                    SENDER => contract_package_hash,
+                    RECIPIENT  => recipient,
+                    TOKEN_IDS => token_ids,
+                },
+            );
+        } else {
+            // CEP-78
+            let token_hash = base16::encode_lower(&runtime::blake2b(&token_id));
```

Comment:
This seems weird. You should just send the string of the token id, which you should already know. The byte encoding of a blake2b hash is how that string is generated in the first place, from the metadata, in CEP-78.

Reply:
Thanks this commit shoud fix that https://github.com/CasperLabs/casper-private-auction/pull/23/commits/3a56e6d49aced55900577114ea7cc06b1fd87c86

I set a const for the token hash, tell me if ok for you

https://github.com/CasperLabs/casper-private-auction/pull/23/commits/3a56e6d49aced55900577114ea7cc06b1fd87c86#diff-9135625ae4709b36720e878940c3b03ef51eeb47db01d4e51a2fe161c2b1aa75